### ISSUE TITLE
t.words.test_jabberxmlstream flaky with randomized hashes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5753]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5753|
|Type|defect|
|Created|2012-07-08 23:06:35Z|

The following test fails sporadically with ``python -R``:

```
[FAIL]
Traceback (most recent call last):
  File "/home/antoine/twisted/twisted/words/test/test_jabberxmlstream.py", line 92, in testSend
    self.xmlstream.transport.value())
  File "/home/antoine/twisted/twisted/trial/unittest.py", line 270, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = "<iq type='get' id='H_11'/>"
b = "<iq id='H_11' type='get'/>"


twisted.words.test.test_jabberxmlstream.IQTest.testSend
```

Attachments:

* [randomhash_test_jabberxmlstream.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/52c/52c1431011c29892d30e8770cb8196c6d321f3dd/5b73a1c189c220a648b84fc21feacad0ba38a2d5.patch) (731 bytes) - added by antoine on 2012-07-09 18:33:52Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5753 5753
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1341788795000000 1341788795000000
changetime__1342615568000000 1342615568000000
version__None None
owner__antoine antoine

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_2"></a>|@pitrou commented|
|-|-|

Here is a patch relaxing the ordering requirement in the test. It therefore fixes the failure.
(relaxing the ordering requirement sounded simpler and more readable than XML-parsing back the transport's value)

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set owner to @pitrou|
|-|-|

This looks good enough to me. Please a news snippet and this is good to go. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_5"></a>|@pitrou commented|
|-|-|

Replying to [ralphm](#note_4):
> This looks good enough to me. Please a news snippet and this is good to go. Thanks!

(same as in [#5752](https://github.com/twisted/twisted/issues/5752) :-))
I can't merge since I don't have commit rights. As for the news snippet, I don't think it warrants one: it's just a test fix, not a bug fix.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_6"></a>|@ralphm set status to `closed`|
|-|-|

(In [34881]) Apply randomhash_test_jabberxmlstream.patch: adjust test to consider nondeterministic order of XML attributes.

Author: antoine.
Reviewer: ralphm.
Fixes: [#5753](https://github.com/twisted/twisted/issues/5753).
